go.mod

```diff
@@ -5,6 +5,7 @@ go 1.21.5
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/blang/semver/v4 v4.0.0
+	github.com/creasty/defaults v1.7.0
```

Comment:
I would rather use mergo for this, the default would just be a struct instead of all these annotations

Reply:
My thought using this would be to also set the jsonschema generated defaults by reading these tags. jsonschema also has tags for defaults, so may ultimately drop this library if it winds up being difficult to use these for both purposes.